### PR TITLE
Fix object and normalization based on new spec

### PR DESCRIPTION
**Pull request by acbramley**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -280,20 +280,20 @@ class Client implements ClientInterface {
     'schedule' => [
       'get' => [
         'action' => 'GET',
-        'uri'    => '/apis/extension.shepherd/v1/namespaces/{namespace}/schedules/{name}',
+        'uri'    => '/apis/extension.shepherd/v1/namespaces/{namespace}/backupscheduleds/{name}',
```
Comment (singularo) [inline src/Client.php:283]: Err bit awkward naming. Shouldn't it be 
scheduledbackups or
backupschedules ?

Comment (acbramley) [reply src/Client.php:283]: Yeah it's what kubebuilder spat out, I'll see if @nicksantamaria can get it changed. We'd need to do it before the crds are installed though.

File: src/Serializer/BackupNormalizer.php [hunk 2]
```diff
@@ -25,23 +24,13 @@ public function denormalize($data, $class, $format = NULL, array $context = [])
...
-      ->setCreationTimestamp($data['metadata']['creationTimestamp']);
-    if (isset($data['metadata']['annotations'])) {
-      $backup->setAnnotations($data['metadata']['annotations']);
-    }
+      ->setCreationTimestamp($data['metadata']['creationTimestamp'])
+      ->setAnnotations($data['metadata']['annotations'] ?? [])
+      ->setPhase($data['status']['phase'] ?? '')
+      ->setStartTimeStamp($data['status']['startTime'] ?? '')
+      ->setCompletionTimeStamp($data['status']['completionTime'] ?? '')
+      ->setResticId($data['status']['resticId'] ?? '');
```
Comment (pingers) [inline src/Serializer/BackupNormalizer.php:32]: feels good

File: src/Client.php (hunk 1 above)
Comment (nicksantamaria) [reply src/Client.php:283]: I went back and forth a bit on the naming - landed on `BackupScheduled` for 2 main reasons.

1. Code organisation in `shepherd-operator` is a bit nicer, with backup related files all grouped together alphabetically.
2. It keeps the CRDs grouped together alphabetically when running `oc get` and `oc get crd`

Unfortunately the plural naming is a bit awkward, and we've hit this with other operators too. I prefer to just use the singular.

File: src/Serializer/BackupNormalizer.php (hunk 2 above)
Comment (singularo) [reply src/Serializer/BackupNormalizer.php:32]: hehe

**Review by singularo — APPROVED** [review]
Looks good to me.

**Pull request merged**